source/Halibut/Transport/ConnectionManager.cs

```diff
@@ -58,6 +80,12 @@ public async Task<IConnection> AcquireConnectionAsync(ExchangeProtocolBuilder ex
                 var c = lazyConnection.Value;
             });
         }
+        
+        async Task<IConnection> CreateNewConnectionWithIOAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken)
```

Comment:
IO is intentional since the other calls "Create" a connection but don't actually and do the IO in some other place.